### ISSUE TITLE
Assigning to rvalue (void 0)

### ISSUE THREAD

**Issue by pkese** [fable-compiler/Fable#2539]
### Description

I'm getting this error while trying to run compiled code:
```js
Error:

Module parse failed: Assigning to rvalue (81:9)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|             throw exn;
|         }
>         (void 0) = (void 0);
|         if (!(caller == null)) {
|             void AdaptiveObject__get_Outputs(x_2).Add(caller);
```

### Repro code

Its a bunch of uncomitted code that I'm trying to upgrade to Fable 3, but in essence the source code is this:  
https://github.com/fsprojects/FSharp.Data.Adaptive/blob/8d751c441aa4b43f590692bcd5f527038134137e/src/FSharp.Data.Adaptive/Core/AdaptiveObject.fs#L165
inlined into this:  
https://github.com/krauthaufen/Fable.Elmish.Adaptive/blob/69e56e7fc9a13d4e9bceff043ed1425550af7a60/src/Fable.React.Adaptive/AttributeUpdater.fs#L93

... and it gets compiled into
```js
export function AttributeUpdater__Update_Z5EE28162(x, t) {
    let copyOfStruct;
    const x_1 = x;
    const x_2 = x_1;
    const token_1 = t;
    const caller = token_1.caller;
    const depth = AdaptiveObject_get_UnsafeEvaluationDepth() | 0;
    let res = null;
    try {
        AdaptiveObject_set_UnsafeEvaluationDepth_Z524259A4(depth + 1);
        const r = ((token_2) => {
            if (AdaptiveObject__get_OutOfDate(x_1)) {
                ((t_1) => {
                    AttributeUpdater__perform(x, AttributeMapReader__get_State(x.reader), AbstractReader$1__GetChanges_Z5EE28162(x.reader, t_1));
                })(token_2);
            }
        })((copyOfStruct = token_1, AdaptiveToken__WithCaller_Z627B3707(copyOfStruct, x_2)));
        AdaptiveObject__set_OutOfDate_Z1FBCCD16(x_2, false);
        const maxAllowedLevel = ((depth > 1) ? (Transaction_get_RunningLevel() - 1) : Transaction_get_RunningLevel()) | 0;
        if (AdaptiveObject__get_Level(x_2) > maxAllowedLevel) {
            const exn = new LevelChangedException(AdaptiveObject__get_Level(x_2) + depth);
            throw exn;
        }
        (void 0) = (void 0);
        if (!(caller == null)) {
            void AdaptiveObject__get_Outputs(x_2).Add(caller);
            caller.Level = (max((x_3, y) => comparePrimitives(x_3, y), caller.Level, AdaptiveObject__get_Level(x_2) + 1) | 0);
        }
    }
    catch (matchValue) {
        AdaptiveObject_set_UnsafeEvaluationDepth_Z524259A4(depth);
        throw matchValue;
    }
    AdaptiveObject_set_UnsafeEvaluationDepth_Z524259A4(depth);
    if (((depth === 0) ? (caller == null) : false) ? (!Transaction_get_HasRunning()) : false) {
        AdadptiveObjectExtensions_AfterEvaluateCallbacks_Run();
    }
}
```

I can try to get my code in order and push ti to Github if the above is not helpful enough.

### Related information

* Fable version: 3.3.0
* Operating system: Linux

**Comment by pkese** [comment]
I have managed to isolate the problem:
```fsharp
let inline f (x:'T) = Unchecked.defaultof<'T> |> ignore
//let inline f x = 0 |> ignore
//let inline f x = 42

let inline doit x (f: 'A -> 'T) =
    let mutable r = Unchecked.defaultof<_>
    r <- f x   
    r

printfn "%A" (doit 1 f)
```

The following code in proper dotnet F# runs fine (`doit` returns unit), whereas fable renders
```js
(function () {
    let arg10;
    let r = null;
    (void 0) = (void 0);
    arg10 = (void 0);
    toConsole(printf("%A"))();
})();
```

**Comment by pkese** [comment]
Closed in favor of #2548 with simplified description.
